bears/vcs/git/GitCommitBear.py

```diff
@@ -24,4 +25,27 @@ def get_remotes():
 
     def get_head_commit(self):
         with change_directory(self.get_config_dir() or os.getcwd()):
+            command = self.github_merge_commit()
+            if command:
+                return run_shell_command(command)
+
             return run_shell_command('git log -1 --pretty=%B')
+
+    def github_merge_commit(self):
+        with change_directory(self.get_config_dir() or os.getcwd()):
+            stdout, stderr = run_shell_command('git log -1 --pretty=%B')
+
+            pos = stdout.find('\n')
+            shortlog = stdout[:pos] if pos != -1 else stdout
+
+            magic_github_merge_commit_regex = re.compile(
+                r'^Merge ([0-9a-f]{40}) into ([0-9a-f]{40})$')
+            match = re.fullmatch(magic_github_merge_commit_regex, shortlog)
+
+            if len(shortlog) == 92 and match:
```

Comment:
`len(shortlog) == 92` is useless.
the regex already specifies an exact character count.  It can only be 92.